routes/api/homepage.js

```diff
@@ -0,0 +1,49 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const passport = require('passport');
+const Post = require('../../models/Posts');
+const User = require('../../models/User');
+
+router.get('/home', passport.authenticate('jwt', { session: false }), (req, res) => {
+  var userFollowing = [];
+  var myPost = mongoose.Types.ObjectId(req.user.id);//get logged in user id to get post
+  
+  User.findById(req.user.id, ('following'))
+  .populate('following.user').lean()
+  .then(following => {
+    if (following) {
+      userFollowing = following.following.map(follow => follow.user._id);
+    } 
+    //Pushing userid of logged in user into userFollowing[] to fetch posts
+        userFollowing.push(myPost);
+        console.log(userFollowing); 
+        //
+        Post.find({ "postedBy": { "$in": userFollowing } } ).lean()
+            .populate('postedBy likes.likedBy comments.commentedBy ',
+              ['name', 'username', 'avatar'])
+            .sort({ timePosted: -1 })//for latest records            
+            .exec((err, records) => {
+              if(records) {
```

Comment:
what will happen if there's no record?

Reply:
Let the comments be there in homepage.js. There is another way to do the above logic. so I commented it

Reply:
removed